modules/regular_service_perimeter/main.tf

```
@@ -134,11 +134,11 @@ resource "google_access_context_manager_service_perimeter" "regular_service_peri
               content {
                 service_name = operations.key
                 dynamic "method_selectors" {
-                  for_each = merge(
+                  for_each = operations.key != "*" ? merge(
                     { for k, v in lookup(operations.value, "methods", {}) : v => "method" },
-                  { for k, v in lookup(operations.value, "permissions", {}) : v => "permission" })
+                  { for k, v in lookup(operations.value, "permissions", {}) : v => "permission" }) : {}
                   content {
-                    method     = method_selectors.value == "method" ? method_selectors.key : ""
+                    method     = method_selectors.value == "method" ? method_selectors.key : null
                     permission = method_selectors.value == "permission" ? method_selectors.key : ""
```

nit: Since you changed the above, should this also default to `null` rather than `""`?
```suggestion
                    permission = method_selectors.value == "permission" ? method_selectors.key : null
```